src/Framework/QueryBuilder/Concerns/CRUD.php

```diff
@@ -2,13 +2,24 @@
 
 namespace Give\Framework\QueryBuilder\Concerns;
 
+use Give\Donations\Models\Donation;
+use Give\Donors\Models\Donor;
 use Give\Framework\Database\DB;
+use Give\Framework\Exceptions\Primitives\InvalidArgumentException;
+use Give\Framework\Models\Contracts\ModelCrud;
+use Give\Framework\Models\Model;
+use Give\Subscriptions\Models\Subscription;
 
 /**
  * @unreleased
  */
 trait CRUD
 {
+    /**
+     * @var string
+     */
+    private $model;
+
```

Comment:
I'm not sure I'm a fan of coupling the QB to the models. Where is this Trait used?

Also, if we do go down this path, we can make use of generics: https://blog.jetbrains.com/phpstorm/2021/12/phpstorm-2021-3-release/#infer_generic_types_from_constructor

Reply:
This whole setModel part within the QB was inspired by Laravel, although not the same thing. Basically we needed a way to set the model on the query builder so we can return instances of the Model instead of generic objects that gets runs through each model's implementation of `fromQueryBuilderObject()`.  This cleaned up sooo much redundant code.  You can find these being set in each repository within `prepareQuery()`.

Reply:
Having QB coupled with models would be nice if we could query models by using model properties. The problem with supporting that is our current DB structure. Sure, it is possible to do such thing by adding a significant amount of complexity, but I'm not sure what would be the benefit of doing that. Having these nice and organized models it's easy to forget that most of our model properties are actually representations of **two** table rows in the meta table. The alternative is to move all the _extra_ things we need from our models into the repositories. 

I like what @jonwaldstein is doing here, but I would suggest doing that by using a slightly different approach - extend the QB class and override the `get` and `getAll` methods, or add more specific methods like `getModelName`, `getAllModelName`. QB is meant to be an all-purpose QB used in Give core, our add-ons, 3rd party add-ons, etc. so it might be a good idea to separate these things.

Reply:
@alaca in the CRUD trait i created 2 methods `getAsModel()` and `getAllAsModel()` which does what you are explaining.  They are currently being used inside `get()` & `getAll()` when the model is set to the QB. I went back and forth with using those explicitly and decided the first example seemed redundant. I do see your point about model properties & the database structure.  I prefer how it works now because the fluent api stays within context of the query - but if we all agree it's better to just have separate explicit methods for returning the query as model I can be persuaded.

`Donation::query()->getAllAsModel()`
`$donation->subscriptions()->getAllAsModel()`

vs

`Donation::query()->getAll()`
`$donation->subscriptions()->getAll()`

Reply:
What I suggested wouldn't affect the current API. I'm just suggesting moving the model logic outside the QB into a subclass.

Reply:
Ahh I see what you're saying now, sorry I misunderstood.  Yes, that's another good option.

Reply:
No worries. I probably have to finish one, or two more coffees before writing anything in the morning. Anyway, what you have here is fine, and it will help us to write less code. I'm just thinking how we could, maybe, have this better organized.

Reply:
Okay, let me see what I can do.

Reply:
All set b6f9247c